### PR TITLE
Linter fixes

### LINKED ISSUE #3632: Redundant newHeads Subscriptions when waiting on multiple outstanding requests.
#### Expected behavior
When I'm using a websocket or other subscription-capable provider, I'd assume that a single `newHeads` subscription would be all that's necessary for web3 to be notified when my transaction is confirmed.

#### Actual behavior
When I'm using a websocket or other subscription-capable provider, web3.js sets up one `newHeads` subscription for every outstanding transaction.

#### Steps to reproduce the behavior

1. Send a transaction.
2. Quickly send another one before the previous one is mined.
3. Count the number of `newHeads` subscriptions active on the client (it'll have increased by two).

#### Environment

I'm running into this primarily in a NodeJS environment while performance testing Hyperledger Besu.

Admittedly this likely isn't a major issue for public Ethereum, but it becomes quite substantial when submitting a high volume of transactions to a private network. For example, on a network with a 2s block time (typical for IBFT 2 networks), if the user is sending 100 transactions per second, there will be up to 200 `newHeads` subscriptions active on the client, all from a single user.

### PR DESCRIPTION
**Pull request by benjamincburns**
## Description

In order to fix #3632, I'm planning on doing some work to refactor the transaction execution flow. When I got started on that I noticed a bunch of linter errors when running `npm test`. These make it hard to check that my changes haven't broken anything, so I'm submitting a PR to fix them.

Note that this doesn't fix the linter issues in `web3-core-method` pertaining to deeply nested blocks. Those will be fixed in the PR for #3632.

Note that some tests appear to be failing on my machine when I run `npm run test:unit` and `npm run test:cov`. These tests were failing in `1.x` before I made this change.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network. (N/A - I'm not introducing new logic changes - it passes all automated client tests)
- [ ] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder. (N/A - I'm not introducing new logic changes)

### CODE REVIEW

File: packages/web3-eth-accounts/src/index.js [hunk 1]
```diff
@@ -149,29 +149,7 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
...
-            error = new Error('"gas" is missing');
-        }
-
-        if (tx.nonce < 0 ||
-            tx.gas < 0 ||
-            tx.gasPrice < 0 ||
-            tx.chainId < 0) {
-            error = new Error('Gas, gasPrice, nonce or chainId is lower than 0');
-        }
+        _validateTransactionForSigning(tx);
```
Comment (benjamincburns) [inline packages/web3-eth-accounts/src/index.js]: Forgot to handle the returned error here. Will push a fix shortly.

**Comment by benjamincburns** [comment]
Update: if I switch back to node v10.21.0, tests pass just fine.

**Comment by benjamincburns** [comment]
>I ran npm run build and tested dist/web3.min.js in a browser.

I'd like to check this off, but is there an existing test harness? The application that I'm working on doesn't run in the browser.

If the `e2e_browsers` CI test is passing, I'd assume we can call this good?

**Comment by GregTheGreek** [comment]
The `netlify/web3-staging/deploy-preview` as a part of the CI actually runs web3 in the browser, so it seems to work!

File: packages/web3/src/index.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ var Web3 = function Web3() {
 
     // overwrite package setProvider
     var setProvider = this.setProvider;
-    this.setProvider = function (provider, net) {
```
Comment (GregTheGreek) [inline packages/web3/src/index.js:55]: line 56 `arguments` intakes `net`

**Review by GregTheGreek — CHANGES_REQUESTED** [review]

File: packages/web3/src/index.js (hunk 2 above)
Comment (benjamincburns) [reply packages/web3/src/index.js:55]: The `arguments` keyword contains all arguments passed, regardless of whether args are bound to variable names in the function signature.

A lot of best practices advice written today suggests that [the `arguments` keyword should be avoided](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments#Description). Perhaps this code would be more readable if it were rewritten to use a [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)?

Comment (GregTheGreek) [reply packages/web3/src/index.js:55]: I would agree with this, I think I'll be tackling this issue next week (or this weekend). Theres unfortunately a lot of usages of `arguments` throughout the subpackages which inherintely use `net`. It got implemented like `super` unfortunately...

**Review by GregTheGreek — APPROVED** [review]

**Pull request merged**